src/Service/FirebaseServiceBase.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace Drupal\firebase\Service;
+
+use Drupal\Component\Serialization\Json;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Logger\LoggerChannelFactoryInterface;
+use Drupal\firebase\FirebaseServiceInterface;
+use GuzzleHttp\ClientInterface;
+
+/**
+ * Provides a base class for service, working with FCM.
+ */
+class FirebaseServiceBase implements FirebaseServiceInterface {
+
+  /**
+   * The config factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactory
+   */
+  protected $configFactory;
+
+  /**
+   * HTTP client.
+   *
+   * @var \GuzzleHttp\Client
+   */
+  protected $client;
+
+  /**
+   * The logger Factory.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelFactory
```

Comment:
it is logger channel - not factory `\Drupal\Core\Logger\LoggerChannelInterface` update comment as well